web_ui/server.py

```
@@ -1216,21 +1452,29 @@ def _write_audit_log(action: str, path: list[str], old_value: Any, new_value: An
         # Determine acting user: session -> Basic auth username -> persisted user -> remote_addr
-        persisted = auth_mod.load_user()
-        user = _session_get("username") or (request.authorization.username if request.authorization is not None else None) or (persisted.get("username") if persisted else None) or request.remote_addr
+        persisted = _load_user_record()
+        user = (
+            _session_get("username")
+            or (request.authorization.username if request.authorization is not None else None)
+            or (persisted.get("username") if persisted else None)
+            or request.remote_addr
+        )
         # Redact sensitive fields from values before serializing to the audit log.
+        path_is_sensitive = any(_is_sensitive_key(component) for component in path)
```

Ruff (0.15.21)

[error] Undefined name `_is_sensitive_key`

(F821)